pkg/activator/handler/handler_test.go

```diff
@@ -357,6 +358,58 @@ func TestActivationHandler(t *testing.T) {
 	}
 }
 
+func TestActivationHandlerProbeCaching(t *testing.T) {
+	namespace := testNamespace
+	revName := testRevName
+	revID := activator.RevisionID{Namespace: namespace, Name: revName}
+
+	fakeRT := activatortest.FakeRoundTripper{}
+	rt := network.RoundTripperFunc(fakeRT.RT)
+
+	breakerParams := queue.BreakerParams{QueueDepth: 10, MaxConcurrency: 10, InitialCapacity: 10}
+	reporter := &fakeReporter{}
+
+	throttler := activator.NewThrottler(
+		breakerParams,
+		endpointsInformer(endpoints(namespace, revName, breakerParams.InitialCapacity)),
+		sksLister(sks(namespace, revName)),
+		revisionLister(revision(namespace, revName)),
+		TestLogger(t))
+
+	handler := (New(TestLogger(t), reporter, throttler,
+		revisionLister(revision(namespace, revName)),
+		serviceLister(service(namespace, revName, "http")),
+		sksLister(sks(namespace, revName)),
+	)).(*activationHandler)
+
+	// Setup transports.
+	handler.transport = rt
+	handler.probeTransport = rt
+
+	sendRequest(namespace, revName, handler)
+	if fakeRT.NumProbes != 1 {
+		t.Errorf("NumProbes = %d, want: %d", fakeRT.NumProbes, 1)
+	}
+
+	sendRequest(namespace, revName, handler)
+	// Assert that we didn't reprobe
+	if fakeRT.NumProbes != 1 {
+		t.Errorf("NumProbes = %d, want: %d", fakeRT.NumProbes, 1)
+	}
+
+	// Dropping the capacity to 0 causes the next request to probe again.
+	throttler.UpdateCapacity(revID, 0)
+	time.AfterFunc(100*time.Millisecond, func() {
+		// Asynchronously updating the capacity to let the request through.
+		throttler.UpdateCapacity(revID, 1)
+	})
+
+	sendRequest(namespace, revName, handler)
+	if fakeRT.NumProbes != 2 {
```

Comment:
Do we technically need to `atomic.Load` these? Maybe add an accessor method and make the field private instead?

Reply:
I think `sendRequest` is synchronous.
>go test . -race -count=100
ok      knative.dev/serving/pkg/activator/handler       43.333s